python/cudf/cudf/window/rolling.py

```diff
@@ -164,6 +208,41 @@ def mean(self):
     def count(self):
         return self._apply_agg("count")
 
+    def apply(self, func, *args, **kwargs):
+        """
+        Counterpart of pandas.core.window.Rolling.apply
+
+        *func* is a user defined function that takes an 1D array as input:
+
+        * We are using `numba` CUDA compiler and the supported Python features
+          are listed in
+
+            https://numba.pydata.org/numba-doc/dev/cuda/cudapysupported.html
+
+          with the exception that math functions in `cmath` are not supported
+          since `libcudf` does not have complex number support and output of
+          `cmath` functions are most likely complex numbers
+
+        """
+        has_nulls = False
+        if isinstance(self.obj, cudf.Series):
+            if self.obj._column.null_count > 0:
+                has_nulls = True
+        else:
+            for col in self.obj._cols:
+                if col.null_count > 0:
```

Comment:
I was trying this out where self.obj is a cudf.DataFrame. It seems like `for col in self.obj._cols:` does not return a cudf series. Instead it just returns the column name. You might need to modify the logic here as I get an error `string has no attribute null_count`. You might need to do something like `self.obj[col]._column.null_count`.

Reply:
@ayushdg Yes here I just want to make sure there is no null values in `obj` since we haven't been able to handle this. You can safely comment this part out. For your previous question, currently the UDF will be applied to each of the columns individually.

Reply:
@ayushdg I have changed the python code so now it handles dataframes correctly. I also added a test to test dataframe inputs.